### PR TITLE
feat: LT-1671 - Permissions to create and edit document templates

### PR DESCRIPTION
**Pull request by jfaracik**

### CODE REVIEW

File: conf/constants.py [hunk 1]
```diff
@@ -54,3 +54,12 @@
 
 # ROLE IDS
 SUPER_USER_ROLE_ID = "00000000-0000-0000-0000-000000000002"
+
+
+class Permissions:
```
Comment (a-bek) [inline conf/constants.py]: Shall we change this to an Enum class?

You'll have to update the signature of `def has_permission(permission: str):`

Also must update where you use it, - `if helpers.has_permission(request, permission):` to `permission.value`

File: conf/decorators.py [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+        @wraps(view_func)
+        def _wrapped_view(request, *args, **kwargs):
+            if helpers.has_permission(request, permission):
+                return view_func(request, *args, **kwargs)
 
-            raise Http404()
+            raise PermissionDeniedError(
+                f"You don't have the permission '{permission}' to view this, "
+                "check urlpatterns or the function decorator if you want to change "
+                "this functionality."
```
Comment (markm5A8Z9F43913A) [inline conf/decorators.py]: Content?

Comment (jfaracik) [reply conf/decorators.py]: This is solely for devs, regular users won't see this :)

File: conf/constants.py (hunk 1 above)
Comment (jfaracik) [reply conf/constants.py]: Good suggestion! I've done it :)

File: core/helpers.py [hunk 3]
```diff
@@ -6,4 +11,29 @@ def convert_dict_to_query_params(dictionary):
...
+
+def has_permission(request, permission: Permission):
+    """
+    Returns true if the user has a given permission, else false
+    """
+    user_permissions = get_user_permissions(request)
+    return permission.value in user_permissions
+
+
+def decorate_patterns_with_permission(patterns, permission: Permission):
```
Comment (a-bek) [inline core/helpers.py]: 👍

**Pull request merged**